src/aten/IndexingOps.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/ATen.h>
+#include <ATen/ExpandUtils.h>
+#include <ATen/MemoryOverlap.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/ScalarOps.h>
+#include <ATen/TensorOperators.h>
+#include <ATen/XPUNativeFunctions.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/native/TensorAdvancedIndexing.h>
+#include <ATen/native/TensorIterator.h>
+#include <aten/sycl/IndexingOpsKernels.h>
+#include <torch/library.h>
+
+namespace at {
+
+Tensor& XPUNativeFunctions::masked_fill_(
+    Tensor& self,
+    const Tensor& mask,
+    const Scalar& value) {
+  TORCH_CHECK(
+      self.device() == mask.device(),
+      "expected self and mask to be on the same device, but got mask on ",
+      mask.device(),
+      " and self on ",
+      self.device());
+  TORCH_CHECK(
+      mask.scalar_type() == kBool,
+      "masked_fill only supports boolean masks, but got dtype ",
+      mask.scalar_type());
+  auto maybe_outnames =
+      namedinference::broadcast_to_outnames(self, mask, "masked_fill_");
+  if (at::has_internal_overlap(self) == MemOverlap::Yes) {
+    TORCH_WARN(
+        "Use of masked_fill_ on expanded tensors is deprecated. "
+        "Please clone() the tensor before performing this operation. "
+        "This also applies to advanced indexing e.g. tensor[mask] = scalar");
+  }
+  at::assert_no_partial_overlap(self, mask);
+
+  c10::MaybeOwned<Tensor> b_mask = expand_inplace(self, mask, "masked_fill_");
+
+  auto iter = TensorIteratorConfig()
+                  .set_check_mem_overlap(false)
+                  .check_all_same_dtype(false)
+                  .resize_outputs(false)
+                  .add_output(self)
+                  .add_const_input(self)
+                  .add_const_input(*b_mask)
+                  .build();
+
+  native::xpu::masked_fill_kernel(iter, value);
+  namedinference::propagate_names_if_nonempty(self, maybe_outnames);
+  return self;
+}
+
+Tensor& XPUNativeFunctions::masked_fill_(
+    Tensor& self,
+    const Tensor& mask,
+    const Tensor& value) {
+  TORCH_CHECK(
+      value.dim() == 0,
+      "masked_fill_ only supports a 0-dimensional value tensor, but got tensor "
+      "with ",
+      value.dim(),
+      " dimension(s).");
+  TORCH_CHECK(
+      !self.device().is_cpu(),
```

Comment:
Looks we should check is_xpu, it will be better, even we have aligned with CUDA.